overlord/devicestate/devicemgr.go

```diff
@@ -723,12 +726,75 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
 	return findSerial(m.state, nil)
 }
 
+type SystemAction struct {
+	Title string
+	Mode  string
+}
+
+type System struct {
+	// Current is true when the system running now was installed from that
+	// seed
+	Current bool
+	// Label of the seed system
+	Label string
+	// Model assertion of the system
+	Model *asserts.Model
+	// Brand information
+	Brand *asserts.Account
+	// Actions available for this system
+	Actions []SystemAction
+}
+
+func systemSeedModelAndBrand(label string) (model *asserts.Model, brand *asserts.Account, err error) {
+	s, err := seed.Open(dirs.SnapSeedDir, label)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot open: %v", err)
+	}
+	if err := s.LoadAssertions(nil, nil); err != nil {
+		return nil, nil, fmt.Errorf("cannot load assertions: %v", err)
+	}
+	// get the model
+	model, err = s.Model()
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot obtain model: %v", err)
+	}
+	brand, err = s.Brand()
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot obtain brand: %v", err)
+	}
+	return model, brand, nil
+}
+
 // Systems list the available recovery/seeding systems.
-func (m *DeviceManager) Systems() ([]string, error) {
-	// TODO:UC20 list available systems in the seed, load each with
-	// seed.LoadAssertions()
-	// TODO:UC20 convert brand-id to user friendly brand name
-	return nil, fmt.Errorf("not implemented")
+func (m *DeviceManager) Systems() ([]System, error) {
+	systemLabels, err := filepath.Glob(filepath.Join(dirs.SnapSeedDir, "systems", "*"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			// maybe not a UC20 system
+			return nil, nil
+		}
+		return nil, fmt.Errorf("cannot list available systems: %v", err)
+	}
+	var systems []System
+	for _, fpLabel := range systemLabels {
+		label := filepath.Base(fpLabel)
+		model, brand, err := systemSeedModelAndBrand(label)
+		if err != nil {
+			// skip broken seed?
```

Comment:
we might have a Broken string in the result like for snaps (see snap.Info.Broken?) ? maybe leave a TODO about that